x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/JdbcTestUtils.java

```diff
@@ -140,4 +160,95 @@ public static void logLikeCLI(ResultSet rs, Logger logger) throws SQLException {
     public static String of(long millis, String zoneId) {
         return StringUtils.toString(ZonedDateTime.ofInstant(Instant.ofEpochMilli(millis), ZoneId.of(zoneId)));
     }
-}
+
+    /**
+     * Returns the classpath resources matching a simple pattern ("*.csv").
+     * It supports folders separated by "/" (e.g. "/some/folder/*.txt").
+     * 
+     *  Currently able to resolve resources inside the classpath that are stored in folders on the file-system,
+     *  and inside jars.
+     */
+    public static List<URL> classpathResources(String pattern) throws Exception {
+        ClassLoader cl = JdbcTestUtils.class.getClassLoader();
+
+        while(pattern.startsWith("/")) {
+            pattern = pattern.substring(1);
+        }
+        
+        Tuple<String, String> split = pathAndName(pattern);
+
+        // the root folder searched inside the classpath - default is the root classpath
+        // default file match
+        final String root = split.v1();
+        final String filePattern = split.v2();
+        
+        List<URL> resources = null;
+
+        if (cl instanceof URLClassLoader) {
+            resources = asList(((URLClassLoader) cl).getURLs());
+        } else {
+            // fallback in case of non-standard CL
+            resources = list(cl.getResources(root));
+        }
+
+        List<URL> matches = new ArrayList<>();
+
+        for (URL resource : resources) {
+            String protocol = resource.getProtocol();
+            URI uri = resource.toURI();
+            Path path = PathUtils.get(uri);
+
+            if ("file".equals(protocol) == false) {
+                throw new IllegalArgumentException("Unsupported protocol " + protocol);
+            }
+
+            // check whether we're dealing with a jar
+            // Java 7 FileSystem can be used by it's not that efficient, memory wise
+            if (path.toString().endsWith(".jar")) {
```

Comment:
Are we actually planning on using jar files? If not, isn't this too much?

Reply:
Actually, I wouldn't make the method `public`, but `protected`, trim it down to what the tests classes are going to use it for (reading `*.csv` files on the file system) and not worry about jars. Whenever (if) we are going to read csv files from jars, I would consider then this functionality.

Reply:
Except gradle uses jars from its classpath while Eclipse use folders. The first iteration focused just on that but then the build was failing...

Reply:
Maybe add a comment in code to explain the jars handling decision?